datashield/src/main/java/org/molgenis/datashield/service/DataShieldExpressionRewriterImpl.java

```diff
@@ -0,0 +1,114 @@
+package org.molgenis.datashield.service;
+
+import java.io.StringReader;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.PostConstruct;
+import org.molgenis.datashield.exceptions.DataShieldExpressionException;
+import org.molgenis.r.RConnectionFactory;
+import org.molgenis.r.model.Package;
+import org.molgenis.r.service.PackageService;
+import org.obiba.datashield.core.DSEnvironment;
+import org.obiba.datashield.core.DSMethodType;
+import org.obiba.datashield.core.impl.PackagedFunctionDSMethod;
+import org.obiba.datashield.r.expr.DataShieldGrammar;
+import org.obiba.datashield.r.expr.ParseException;
+import org.obiba.datashield.r.expr.RScriptGenerator;
+import org.rosuda.REngine.REXPMismatchException;
+import org.rosuda.REngine.Rserve.RConnection;
+import org.rosuda.REngine.Rserve.RserveException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DataShieldExpressionRewriterImpl implements DataShieldExpressionRewriter {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(DataShieldExpressionRewriterImpl.class);
+  private final PackageService packageService;
+  private final RConnectionFactory rConnectionFactory;
+
+  private final DSEnvironment aggregateEnvironment;
+  private final DSEnvironment assignEnvironment;
+
+  private final RScriptGenerator rAggregateScriptGenerator;
+  private final RScriptGenerator rAssignScriptGenerator;
+
+  public DataShieldExpressionRewriterImpl(
```

Comment:
This class does a bit much, both keeping track of the methods and wiring all the parts together.